### PR TITLE
abstractions and specifications

### PR DESCRIPTION
**Pull request by bennycio**
Changes:

- **Added** database abstractions so that we are not bound to mongo
- **Removed** runtime package in exchange for a more idiomatic go approach that lets us see more specifically what dependencies any given function/endpoint has

### CODE REVIEW

File: memstore/redis.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	"context"
+	"errors"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+)
+
+const (
+	// CollUnassignedRDB is a throttled cache for expensive queries finding unassigned NFTs
+	CollUnassignedRDB DB = iota
```
Comment (Robinnnnn) [inline memstore/redis.go:13]: why is there a space between `DB DB` ?

File: persist/mongodb/nft.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+	return mongo.Pipeline{
+		{{Key: "$match", Value: matchFilter}},
+		{{Key: "$lookup", Value: bson.M{
+			"from":         "history",
+			"localField":   "_id",
+			"foreignField": "nft_id",
+			"as":           "ownership_history",
+		}}},
+		{{Key: "$set", Value: bson.M{"ownership_history": bson.M{"$arrayElemAt": []interface{}{"$ownership_history", 0}}}}},
+		// {{Key: "$lookup", Value: bson.M{
```
Comment (Robinnnnn) [inline persist/mongodb/nft.go:262]: saving this for later?

File: persist/nft.go [hunk 3]
```diff
@@ -144,250 +130,17 @@ type UpdateNFTInfoInput struct {
...
-		// 			"$expr": bson.M{
-		// 				"$in": []interface{}{bson.M{"$first": "$addresses"}, "$$owners"},
-		// 			},
-		// 		}}},
-		// 	},
-		// 	"as": "owner_users",
-		// }}},
-	}
+// NFTRepository represents the interface for interacting with persisted NFTs
+type NFTRepository interface {
```
Comment (Robinnnnn) [inline persist/nft.go:134]: lovely

File: persist/mongodb/nft.go (hunk 2 above)
Comment (bennycio) [reply persist/mongodb/nft.go:262]: yes! this is from when we made it not fill out username for an NFT.

**Pull request merged**